lmp/model/_transformer.py

```diff
@@ -0,0 +1,684 @@
+r"""Transformer Language Model with nn.TranformerEncoderlayer."""
+
+
+import argparse
+import math
+from typing import ClassVar, Dict, Optional
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from lmp.model._base import BaseModel
+from lmp.tknzr._base import BaseTknzr
+
+
+class PositionalEncoding(nn.Module):
+    r"""Positional Encoding.
+
+    In order for Transformer model to make use of postional information of
+    input sequence, we usually add positional encodings to the input
+    embeddings.
+
+    .. math::
+
+        \begin{align}
+        \text{PE}_{(\text{pos},\text{2i})} &= \sin\
+            (\frac{\text{pos}}{10000^\frac{\text{2i}}{d_{\text{emb}}}}) \\
+        \text{PE}_{(\text{pos},\text{2i+1})} &= \cos\
+            (\frac{\text{pos}}{10000^\frac{\text{2i}}{d_{\text{emb}}}}) \\
+        \end{align}
+
+    Parameters
+    ==========
+    d_emb: int
+        embden dimension for MLP and Transformer.
```

Comment:
The first letter of a sentence should be a capital letter. And it should be `Embedding` instead.